self-core-impl/src/main/java/com/selfxdsd/core/issues/GithubIssues.java

```diff
@@ -63,7 +78,24 @@ public GithubIssues(final URI issuesUri, final Storage storage) {
 
     @Override
     public Issue getById(final int issueId) {
-        throw new UnsupportedOperationException("Not yet implemented.");
+        Issue issue = cachedIssues.get(issueId);
+        if (issue == null) {
```

Comment:
@criske No need to cache the Issue. It is also not correct to cache it since this Issue is live on Github/GitLab etc and could have it's state (e.g. assignee) changed at any time.

Just fetch the Json and return a ``GithubIssue`` or null if the HTTP call returns 404 NOT FOUND or 204 NO CONTENT.

Reply:
Ok

Reply:
One question, what should we return when we get other codes (like 500s) or an IOException? We throw?

Reply:
@criske wrap it inside ISE with a meaningful message and throw it. Similarly to this: https://github.com/self-xdsd/self-core/blob/master/self-core-impl/src/main/java/com/selfxdsd/core/BaseRepo.java#L109

Reply:
> @criske No need to cache the Issue. It is also not correct to cache it because this Issue is live on Github and could have its state (e.g. assignee) changed at any time.
>
Just got an idea about caching, for the future: we could with etags if an issue has changed. I'm pretty sure that github supports etags.

Reply:
@criske maybe, but I'm not sure it would help a lot. We'll see :D